Data/ByteString/Lazy.hs

```diff
@@ -609,11 +620,43 @@ scanl
     -- ^ input of length n
     -> ByteString
     -- ^ output of length n+1
-scanl f z = snd . foldl k (z,singleton z)
- where
-    k (c,acc) a = let n = f c a in (n, acc `snoc` n)
+scanl function = fmap (uncurry (flip snoc)) . mapAccumLChunks (S.mapAccumL (\x y -> (function x y, x)))
```

Comment:
We might not be happy about either solution.
* The former one is not lazy. _(See #372.)_
* The new one uses `snoc` which means going along the spine one more time. I am not sure whether this is a valid concern, since `snoc` is lazy as well. I do not know enough about performance to decide whether I should be worried.

Reply:
I think this is fine, because `snoc` is lazy.

Reply:
Cheers.